RollingStockOwnership/Equipment.cs

```diff
@@ -414,7 +428,7 @@ public float SquaredDistanceFromPlayer()
 
 		// Train car position appears to be world absolute, so we have to compare to the player's world absolute position
 		// This is different from what UnusedTrainCarDeleter appears to be doing, but I'm not sure why
-		return (position - PlayerManager.GetWorldAbsolutePlayerPosition()).sqrMagnitude;
+		return (position - (PlayerManager.PlayerTransform.position - WorldMover.currentMove)).sqrMagnitude;
```

Comment:
I prefer the descriptiveness and conciseness of the old `PlayerManager` method. Let's turn this into a private helper method or a `PlayerManager` extension method.